### DISCUSSION TITLE
How to make connection being connected forever?

### DISCUSSION THREAD

**Discussion by ami-GS** [microsoft/msquic#2185] [Q&A] [answered]
There is IdleTimeoutMs setting, the type is uint64_t.
Is it possible to make the connection forever?
like setting IdleTimeoutMs = 0?
Or
IdleTimeoutMs = UINT64_MAX to theoretically represent forever?

**Comment by nibanks** [comment, marked as answer]
A value of `0` for `IdleTimeoutMs` indicate "no idle timeout", effectively disabling idle timeouts. But be aware, QUIC negotiates idle times for a connection as the minimum of what the client and server offer. So if you disable them on the client side, but the server still has it configured, it will be whatever the server configured it to be.

**Reply by Wizmann** [reply]
What does the "idle" mean for MsQuic?

What will happen if:

* "keep alive" heartbeat has been enabled
* the connection is good but no actual data has been sent (for a long time)

**Reply by nibanks** [reply]
> What does the "idle" mean for MsQuic?

Idle means no ACK-eliciting packet has been sent or received.

> "keep alive" heartbeat has been enabled

Keep alives are ACK-eliciting by design.

> the connection is good but no actual data has been sent (for a long time

How do you actually know a connection is "good" if you haven't sent or received anything on it?

**Reply by Wizmann** [reply]
I understand. Thank you.

> How do you actually know a connection is "good" if you haven't sent or received anything on it?

It's just a hypothesis...

**Comment by ami-GS** [comment]
Thanks!
That solves my problem😋
